BASE/src/Microsoft.ApplicationInsights/Channel/InMemoryChannel.cs

```diff
@@ -186,6 +187,16 @@ public void Flush(TimeSpan timeout)
             }
         }
 
+        /// <summary>
+        /// Will send all the telemetry items stored in the memory asynchronously.
+        /// </summary>
+        public async Task<bool> FlushAsync()
+        {
+            return await Task.Run(() => this.Flush(default(TimeSpan)))
```

Comment:
I don't understand the point of the ContinueWith here.
Consider this (note: no async, no await):
```csharp
public Task<bool> FlushAsync()
{
    return Task.Run(() => this.Flush(default(TimeSpan)));
}
```